tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -5371,6 +5385,29 @@ void CGMSHLSLRuntime::EmitHLSLFlatConversionAggregateCopy(CodeGenFunction &CGF,
       unsigned sizeDest = TheModule.getDataLayout().getTypeAllocSize(DestPtrTy);
       CGF.Builder.CreateMemCpy(DestPtr, SrcPtr, std::min(sizeSrc, sizeDest), 1);
       return;
+    } else if (GlobalVariable *SrcGV = dyn_cast<GlobalVariable>(SrcPtr)) {
+      if (m_ConstVarAnnotationMap.count(SrcGV) &&
+          IsVec4ArrayToScalarArrayForMemcpy(SrcPtrTy, DestPtrTy)) {
+        unsigned sizeSrc = TheModule.getDataLayout().getTypeAllocSize(SrcPtrTy);
+        unsigned sizeDest =
+            TheModule.getDataLayout().getTypeAllocSize(DestPtrTy);
+        if (sizeSrc == sizeDest) {
+          CGF.Builder.CreateMemCpy(DestPtr, SrcPtr, std::min(sizeSrc, sizeDest),
```

Comment:
Why do `std::min(sizeSrc, sizeDest)` when `sizeSrc == sizeDest`?  Same for the next case.